### DISCUSSION TITLE
A guide: ``latex2e to latex3''

### DISCUSSION THREAD

**Discussion by Qinka** [latex3/latex3#1496] [Q&A]
Is there any guide for migrating latex2e to latex3?

For example, I want to know the latex3 version of command `\thispagestyle` and a latex3 style to define a page style.
For latex2e, it looked like:
```latex
\def\ps@xdu@empty{%
  \let\@oddhead\@empty%
  \let\@evenhead\@empty%
  \let\@oddfoot\@empty%
  \let\@evenfoot\@empty}
```
For latex3, ?

**Comment by josephwright** [comment]
There is no separate LaTeX3 - it's a programming layer and philosophy for improvements to the kernel. As such, some things don't have a 'LaTeX3 equivalent' - here I'd likely stick with the current code unless you are doing stuff programmatically.

**Comment by davidcarlisle** [comment]
As Joseph says there isn't really any L3 equivalent here. You could replace `\def` and `\let`  and `\@empty` but you can not change the names `\@oddhead` etc as they are the command names actually called by the LaTeX output routine.  So there isn't really any advantage here.

It might be that you were designing a new page style making use of l3coffins or l3draw or similar modules to position the elements, so it can make sense to write new page styles using L3, but that is making use of L3-specific functionality, not an example of "converting" from latex2e conventions to L3. And even then, at some point the L3 definition of the new style would need to call `\@oddhead`
